UserDataProcessingTrigger/index.ts

```diff
@@ -0,0 +1,69 @@
+import { Context } from "@azure/functions";
+import * as df from "durable-functions";
+import { UserDataProcessingChoiceEnum } from "io-functions-commons/dist/generated/definitions/UserDataProcessingChoice";
+import { UserDataProcessingStatusEnum } from "io-functions-commons/dist/generated/definitions/UserDataProcessingStatus";
+import { UserDataProcessing } from "io-functions-commons/dist/src/models/user_data_processing";
+import * as t from "io-ts";
+import { readableReport } from "italia-ts-commons/lib/reporters";
+
+const logPrefix = "UserDataProcessingTrigger";
+
+// models the subset of UserDataProcessing documents that this orchestrator accepts
+export type ProcessableUserDataDownload = t.TypeOf<
+  typeof ProcessableUserDataDownload
+>;
+export const ProcessableUserDataDownload = t.intersection([
+  UserDataProcessing,
+  // ony the subset of UserDataProcessing documents
+  // with the following characteristics must be processed
+  t.interface({
+    choice: t.literal(UserDataProcessingChoiceEnum.DOWNLOAD),
+    status: t.literal(UserDataProcessingStatusEnum.PENDING)
+  })
+]);
+
+const CosmosDbDocumentCollection = t.readonlyArray(t.readonly(t.UnknownRecord));
+type CosmosDbDocumentCollection = t.TypeOf<typeof CosmosDbDocumentCollection>;
+
+interface ITaskDescriptor {
+  orchestrator: string;
+  input: ProcessableUserDataDownload;
+}
+
+export default (context: Context, input: unknown) => {
+  const tasksDescriptors = CosmosDbDocumentCollection.decode(input)
+    .getOrElseL(err => {
+      throw Error(`${logPrefix}: cannot decode input [${readableReport(err)}]`);
+    })
+    .reduce(
+      (tasks, maybeProcessable) => {
+        if (ProcessableUserDataDownload.is(maybeProcessable)) {
+          return [
+            ...tasks,
+            {
+              input: maybeProcessable,
+              orchestrator: "UserDataDownloadOrchestrator"
+            }
+          ];
+        } else {
+          context.log.warn(
+            `${logPrefix}: skipping document [${JSON.stringify(
+              maybeProcessable
+            )}]`
+          );
+          return tasks;
+        }
+      },
+      [] as readonly ITaskDescriptor[]
+    );
+
+  context.log.info(
+    `${logPrefix}: processing ${tasksDescriptors.length} document${
+      tasksDescriptors.length === 1 ? "" : "s"
+    }`
+  );
+
+  tasksDescriptors.forEach(task =>
+    df.getClient(context).startNew(task.orchestrator, undefined, task.input)
```

Comment:
i think that we need to await Promise.all here otherwise the process will end before promises are settled